### DISCUSSION TITLE
How to uninstall Packs?

### DISCUSSION THREAD

**Discussion by nbro** [symfony/symfony#54916] [Q&A]
Packs is like collection of other packages or bundles. I installed `symfony/serializer-pack` with `composer remove symfony/serializer-pack`. What if I want now to remove this pack, so all the packages that it was installed (both from the lock file and the environment, of course)? `composer remove symfony/serializer-pack` tells me that there's nothing to update or remove

**Comment by MatTheCat** [comment]
Indeed you cannot uninstall packs since they are automatically unpacked (see https://github.com/symfony/flex/issues/645#issuecomment-655677296).

Instead, you need to uninstall each of their dependencies which you can find on packagist. E.g. from https://packagist.org/packages/symfony/serializer-pack the serializer-pack requires

- phpdocumentor/reflection-docblock
- phpstan/phpdoc-parser
- symfony/property-access
- symfony/property-info
- symfony/serializer

so you’d have to run `composer remove phpdocumentor/reflection-docblock phpstan/phpdoc-parser symfony/property-access symfony/property-info symfony/serializer`.

**Reply by nbro** [reply]
If all it takes to uninstall a pack is to uninstall all the packages in the `requires` section, then I wonder why there isn't such a feature in `composer` or in any plugin composer that installs `packs`, which I assume is Flex. Also, if that's not available, then there should be a way to easily extract these `requires` packages of a pack and pass them to `composer remove`. We can probably do that already using some shell (e.g. something similar to what you provided here https://github.com/symfony/symfony/discussions/54917), but why not make this easy?

**Reply by MatTheCat** [reply]
> why not make this easy?

Could be because this is an edge-case, not a big deal, nobody asked… You can always try contributing to [Flex](https://github.com/symfony/flex) and see how it goes.
